### ISSUE TITLE
toggle_ha_active and activate don't update live device

### ISSUE THREAD

**Issue by patrickdaj** [PaloAltoNetworks/pan-os-python#186]
## Describe the bug
Perhaps I am missing the point of these methods but they appear to only update self._ha_active and that state is never transmitted to the actual devices.  I am assuming it is waiting on the unimplemented update_ha_active method?  If not these are broken.  :) . 

## Expected behavior

The activate() method should cause the device to become active.  The toggle_ha_active, should flip the active device to passive and vice versa.

## Current behavior

toggle_ha_active() and activate() only affect the state of the PanDevice but not the live device it is tied to.

## Possible solution
Implement update_ha_active() method and make calls to it from activate() and toggle_ha_active().  I'm willing to fix and submit PR but I wanted to make sure I was not missing something and that was the intent of that unimplemented method.

## Steps to reproduce

fw1 = PanDevice('fw1.local', 'admin', 'admin') # assumed to be primary
fw2 = PanDevice('fw2.local', 'admin', 'admin')

fw1.set_ha_peers(fw2)

fw1.toggle_ha_active()
assert(fw1.is_active() != True) # should return False?, device should not be active

#### Same thing for activate()
fw2.activate()
assert(fw2.is_active() == True)

## Screenshots

N/A

## Context

Having a simple way to toggle HA state is very useful for things like automated upgrades and other testing scenarios I use regularly.

## Your Environment

I tested this against VM-100 and also PA-7080 firewalls running 8.0.20.

**Comment by shinmog** [comment]
I believe this is as designed - @btorresgil can confirm

**Comment by btorresgil** [comment]
The names of these methods are not so clear, so I understand the confusion. As you point out, these methods do not affect the live device. They are part of an internal state machine that allows you to simplify an HA pair into a single firewall and pandevice will always talk to the active firewall in the pair automatically.  This system is not well documented, partly because it's supposed to be seamless to the developer, but I'll try to explain here.

For working with an HA pair, I recommend steps like this:

```python
# Don't assume either firewall is primary or active.
# Any such assumption would be unsafe in the context of an App built to manage firewalls
# that knows nothing about the firewalls ahead of time.
fw = PanDevice('fw1.local', 'admin', 'admin')
fw.set_ha_peers(PanDevice('fw2.local', 'admin', 'admin'))

# Notice I didn't save the second firewall to a variable, because I don't need it.
# The point is to treat the HA pair as one firewall, so we only need one variable.
# This way, we have only one pandevice configuration tree to manage,
# NOT one tree for each fw in the pair.

# At this point, it's a good idea to collect the active/passive state from
# the live devices.
fw.refresh_ha_active()

# And verify the config is synced between the devices.
# If it's not synced, force config synchronization from active to standby
if not fw.config_synced():
    fw.synchronize_config()  # blocks until synced or error

# Now, it's completely safe to use all the configuration methods as usual
# on the one fw variable.
obj = AddressObject('test', '10.0.1.1')
fw.add(obj)
obj.create()
```

Some advantages of this design:

* Only one configuration tree to manage
* Pandevice automatically knows which firewall to talk to
* Pandevice detects when a firewall is not reachable and automatically switches to the other firewall instead.
* Pandevice knows that some configuration should be made on the active fw, and some config should be made on both firewalls. It handles this for you under the hood.

There is currently no convenience method for switching which firewall is active. Since pandevice is primarily a configuration tool, and this operations is not configuration related. But, you could easily do this with an op command:

```python
if fw.is_active():
    fw.op("request high-availability state suspend")
    fw.toggle_ha_active() # or, wait a bit and do `fw.refresh_ha_active()`
```

**Comment by btorresgil** [comment]
Also, in case you're interested, here's the code that does automatic handling of failed firewalls and configuration that needs to be applied on both firewalls.  Again, this is all under the hood, so the developer doesn't need to worry about it. From the developer perspective, they just modify the configuration and (unless both firewalls are down) it should just work.

https://github.com/PaloAltoNetworks/pandevice/blob/d7f675ff5ff50338aa15c4c9420ff82f782ea377/pandevice/base.py#L3415

**Comment by btorresgil** [comment]
I created a PR to add this information to the docs as a How-to Guide for HA Pairs
https://github.com/PaloAltoNetworks/pandevice/pull/193/files#diff-ab69f9275951af7789c4d014c906de1dR154

I would welcome any feedback!

**Comment by btorresgil** [comment]
Fixed via documentation in #193
